### PR TITLE
Add macro commands

### LINKED ISSUE #7: Add command for message macros
Add a command that can save a message for later reuse.

I imagine it to work like this:

To save a message as a macro
- Copy the message ID or message link to save as a macro
- Enter `/macro name:MACRO_NAME message:MESSAGE_ID_OR_LINK` to save the message as a macro
- When a message contains an attachment, print an error, i.e. only store text as macros, not images

To write the contents of a macro (saved message):
- Enter `macro name:MACRO_NAME`
- The command must suggets all stored names for the `name` parameter

To delete a macro:
- Enter `/macro name:MACRO_NAME delete:True`
- This can be the same command as the previous one, with `delete` being optional and `False` by default

This would close #4.

### PR DESCRIPTION
**Pull request by mercurialworld**
Closes #7.

Adds a group of commands with the prefix `macros`:

- `/macros create` (`name`: string): Opens a modal which lets you create an embed. Lets you preview the macro before it's saved.

The macro creation modal.
The preview of the created macro.

- `/macros edit` (`name`: string): The same as `/macros create`, but lets you edit an existing macro.
- `/macros show` (`name`: string, `mention`: (optional) Discord user): Lets you send a macro, and optionally, mention a user.

Showing a macro

- `/macros list`: Lists all created macros.
- `/macros delete` (`name`: string): Deletes a macro. 

There are probably a few edge cases I forgot, so please test this carefully before merging!

### CODE REVIEW

**Comment by mercurialworld** [comment]
Now that I think about it, I could make `/macros show` its own command, `/macro`, with the same arguments... Let me do that real quick.

File: bot.py [hunk 1]
```diff
@@ -421,4 +437,119 @@ async def command_tester_coverage(interaction: discord.Interaction):
...
+    if current is None or current == "":
+        return [
+            discord_command.Choice(name=macro_name[0], value=macro_name[0])
+            for macro_name in macro_names(bot_utils.macros_db)
+        ]
+    else:
+        return [
+            discord_command.Choice(name=macro_name[0], value=macro_name[0])
+            for macro_name in macro_search(bot_utils.macros_db, current)
+        ]
```
[the remark this answers is not in the document]
Comment (mercurialworld) [reply bot.py:549]: i see your point, but i'm unsure how i would want to go about caching these... probably just have an in-memory list of just the macro names (sorted by date edited) and then put it through some sort of fuzzy search? 

this would mean that on every macro created/edited i'd have to run something to update the array... could be another PR

File: enums/constants.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 ROLE_BETA_TESTER = 1349699182968967219
 ROLES_OS = [1295480990722035752, 1295480950242676737, 1295480841987821628]
+ROLES_USE_MACROS = ["Moderator"]
```
[the remark this answers is not in the document]
Comment (mercurialworld) [reply enums/constants.py]: > Scratch that, it's possible to limit commands to specific roles in the Discord server settings, so you do not need to do this explicitly.

oh, that's actually pretty cool! i've removed the check - it was mainly a very dirty bandage patch to prevent abuse but because that's possible this is no longer needed :>

**Review by ungive — APPROVED** [review]
LGTM! Thank you for your contribution!
